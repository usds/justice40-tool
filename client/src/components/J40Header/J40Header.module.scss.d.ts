declare namespace J40HeaderNamespace {
    export interface IDatasetCardScss {
        fullScreenContainer: string;
        bannerContainer: string;
        language: string;
        logoNavRow: string;
        logo: string;
        logoTitle: string;
        title2BetaPill: string;
        betaPill: string;
        navLinks: string;
        alert: string;
    }
  }

declare const DatasetCardScssModule: J40HeaderNamespace.IDatasetCardScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: J40HeaderNamespace.IDatasetCardScss;
  };

  export = DatasetCardScssModule;
