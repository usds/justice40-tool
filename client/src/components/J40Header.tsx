import React, {useState} from 'react';
import {Link, useIntl} from 'gatsby-plugin-intl';
import {
  Header,
  NavMenuButton,
  PrimaryNav,
  GovBanner,
  Tag,
} from '@trussworks/react-uswds';

// @ts-ignore
import siteLogo from '../../src/images/icon.png';
import * as COMMON_COPY from '../data/copy/common';

const J40Header = () => {
  const intl = useIntl();
  const [mobileNavOpen, setMobileNavOpen] = useState(false);

  const titleL1 = intl.formatMessage(COMMON_COPY.HEADER.TITLE_LINE_1);
  const titleL2 = intl.formatMessage(COMMON_COPY.HEADER.TITLE_LINE_2);

  const toggleMobileNav = (): void =>
    setMobileNavOpen((prevOpen) => !prevOpen);

  const headerLinks = () => {
    // static map of all possible menu items. Originally, it was all strings,
    // but we need to handle both onsite and offsite links.
    const menuData = new Map<string, JSX.Element>([
      ['about',
        <Link
          to={'/'}
          key={'about'}
          activeClassName="usa-current"
          className={'j40-header'}
          data-cy={'nav-link-about'}>
          {intl.formatMessage(COMMON_COPY.HEADER.ABOUT)}
        </Link>,
      ],
      ['cejst',
        <Link
          to={'/cejst'}
          key={'cejst'}
          activeClassName="usa-current"
          className={'j40-header'}
          data-cy={'nav-link-explore'}>
          {intl.formatMessage(COMMON_COPY.HEADER.EXPLORE)}
        </Link>,
      ],
      ['methodology',
        <Link
          to={'/methodology'}
          key={'methodology'}
          activeClassName="usa-current"
          className={'j40-header'}
          data-cy={'nav-link-methodology'}>
          {intl.formatMessage(COMMON_COPY.HEADER.METHODOLOGY)}
        </Link>,
      ],
      ['contact',
        <Link
          to={'/contact'}
          key={'contact'}
          activeClassName="usa-current"
          className={'j40-header'}
          data-cy={'nav-link-contact'}>
          {intl.formatMessage(COMMON_COPY.HEADER.CONTACT)}
        </Link>,
      ],
    ]);

    const menu =['about', 'cejst', 'methodology', 'contact'];
    return menu.map((key) => menuData.get(key));
  };

  return (
    <>
      <GovBanner/>
      <Header
        basic={true} role={'banner'}
        className={'usa-header j40-header'}>
        <div className="usa-nav-container">
          <div className="usa-navbar">
            {/* Removing h1 from logo ease transition to USWDS tokens in headers */}
            {/* https://wehavezeal.com/blog/web-development/2016/01/12/should-i-use-the-h1-tag-for-my-website-logo */}
            <div className="usa-logo">
              <img className="j40-sitelogo" src={siteLogo} alt={`${titleL1} ${titleL2}`} />
              <span className={'usa-logo__text j40-title'}>
                <span className={'j40-title-line1'}>{titleL1}</span><br/>
                <span className={'j40-title-line2'}>{titleL2}</span>
                <Tag className={'j40'}>Beta</Tag>
              </span>
            </div>
            <NavMenuButton
              key={'mobileMenuButton'}
              onClick={toggleMobileNav}
              label="Menu"/>
          </div>
          <PrimaryNav
            items={headerLinks()}
            mobileExpanded={mobileNavOpen}
            onToggleMobileNav={toggleMobileNav}
            className={'j40-header'}
          >
          </PrimaryNav>
        </div>
      </Header>
    </>
  );
};

export default J40Header;
