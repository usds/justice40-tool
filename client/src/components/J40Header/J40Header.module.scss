@use '../../styles/design-system.scss' as *;


.logoNavRow {
  @include u-margin-top(4);

  .logo {
    @include u-width(10);
    @include u-padding("05");
  }

  // The logoTitle declaration is enabled for widths > 1200px
  .logoTitle {
    @include u-display("flex");
    @include u-flex-direction("column");
    @include typeset("sans", 8, 3);
    max-width: 270px;
    min-width: 218px;
  }

  .navLinks {
    @include u-display("flex");
    justify-content: end;
  }

  // This media query limits this declaration to 1024 < width < 1200px
  @include at-media-max("desktop-lg") {
    .logoTitle {
      @include typeset("sans", 7, 3);
    }
  }

  // This media query limits this declaration to 880 < width < 1024px
  @include at-media-max("tablet-lg") {
    .logoTitle {
      @include typeset("sans", 7, 2);
    }
  }

  // This media query limits this declaration to 640 < width < 880px
  @include at-media-max("tablet") {
    .logoTitle {
      @include typeset("sans", 5, 2);
    }

    .betaPill {
      @include u-font("body", "3xs");
    }
  }
}

.alert {
  @include u-margin-top(4);
}