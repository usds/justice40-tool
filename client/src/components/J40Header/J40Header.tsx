import React, {useEffect, useState} from 'react';
import {Link, useIntl} from 'gatsby-plugin-intl';
import {
  Alert,
  Header,
  NavMenuButton,
  PrimaryNav,
  Grid,
  NavDropDownButton,
  Menu,
} from '@trussworks/react-uswds';
import BetaBanner from '../BetaBanner';
import J40MainGridContainer from '../J40MainGridContainer';
import GovernmentBanner from '../GovernmentBanner';
// import Language from '../Language';
import {useWindowSize} from 'react-use';

// @ts-ignore
import siteLogo from '../../images/j40-logo-v2.png';
import * as styles from './J40Header.module.scss';
import * as COMMON_COPY from '../../data/copy/common';
import {PAGES_ENDPOINTS, USWDS_BREAKPOINTS} from '../../data/constants';


interface IJ40Header {
  location: Location
}

/**
 * The J40Header component will control how the header looks for both mobile and desktop
 *
 * The Header is defined as
 * 1. Two rows of Banners (ie, official gov website and beta site)
 * 2. Logo and Nav Links Row
 * 3. Any Alerts
 *
 * @param {Location} location
 * @return {JSX.Element}
 */
const J40Header = ({location}:IJ40Header) => {
  const intl = useIntl();

  // grab last segment of location pathname
  const [lastSegmentLocation] = location.pathname.split('/').slice(-1);

  // Logo text
  const logoLine1 = intl.formatMessage(COMMON_COPY.HEADER.TITLE_LINE_1);

  /**
   * State variable to control the toggling of mobile menu button
   */
  const [mobileNavOpen, setMobileNavOpen] = useState(false);
  const toggleMobileNav = (): void => {
    setMobileNavOpen((prevOpen) => !prevOpen);
  };

  /**
   * State variable to hold the open/close state of each nav dropdown. This will allow for two
   * dropdown that are being used, each corresponding to an index in the state array:
   *
   * index 0 = Data & Methodology dropdown
   * index 1 = About dropdown
   */
  const [isOpen, setIsOpen] = useState([false, false]);

  /**
   * When transitioning between anything larger than desktop and anything less than desktop the nav menu
   * changes from the usual row of nav links (dektop) to the MENU button(mobile). On desktop all nav drop
   * dropdowns should be closed, while on mobile all nav links should be open.
   *
   * The useWindowSize provides the device width and the useEffect allows the side effect (opening/closing
   * nav links) to occur anytime the device width changes.
   */
  const {width} = useWindowSize();
  useEffect( () => {
    if (width < USWDS_BREAKPOINTS.DESKTOP) {
      setIsOpen([true, true]);
    } else {
      setIsOpen([false, false]);
    }
  }, [width]);

  /**
   * This toggle function will handle both navigation toggle links (Meth and About).
   *
   * @param {number} index
   */
  const onToggle = (index: number): void => {
    // The setIsOpen is used to toggle the currently selected nav link
    setIsOpen((prevIsOpen) => {
      const newIsOpen = [...isOpen];
      newIsOpen[index] = !prevIsOpen[index];
      return newIsOpen;
    });

    /**
     * When on desktop only, the dropdown nav links (Meth and About) should close if any of the other ones
     * are still open. This next set of logic handles that.
     */
    if (index === 0 && isOpen[1] === true && width > USWDS_BREAKPOINTS.DESKTOP) {
      setIsOpen([isOpen[0], false]);
    } else if (index === 1 && isOpen[0] === true && width > USWDS_BREAKPOINTS.DESKTOP) {
      setIsOpen([false, isOpen[1]]);
    }
  };

  /**
   * On mobile, the Methodology & Data page should have two sub-nav links. This defines
   * the array that will hold these links
   */
  const methPageSubNavLinks = [
    <Link
      to={PAGES_ENDPOINTS.METHODOLOGY}
      key={'methodology'}
      activeClassName="usa-current"
      data-cy={'nav-link-methodology'}>
      {intl.formatMessage(COMMON_COPY.HEADER.METHODOLOGY)}
    </Link>,
    <Link
      to={PAGES_ENDPOINTS.DOWNLOADS}
      key={'downloads'}
      activeClassName="usa-current"
      data-cy={'nav-link-downloads'}>
      {intl.formatMessage(COMMON_COPY.HEADER.DOWNLOADS)}
    </Link>,
    <Link
      to={PAGES_ENDPOINTS.PREVIOUS_VERSIONS}
      key={'previous-versions'}
      activeClassName="usa-current"
      data-cy={'nav-link-previous-versions'}>
      {intl.formatMessage(COMMON_COPY.HEADER.PREVIOUS_VERSIONS)}
    </Link>,
  ];

  /**
   * On mobile, the About page should have 3 sub-nav links. This defines
   * the array that will hold these links
   */
  const aboutPageSubNavLinks = [
    <Link
      to={PAGES_ENDPOINTS.ABOUT}
      key={'about'}
      activeClassName="usa-current"
      data-cy={'nav-link-about'}>
      {intl.formatMessage(COMMON_COPY.HEADER.ABOUT)}
    </Link>,
    <Link
      to={PAGES_ENDPOINTS.PUBLIC_ENG}
      key={'publicEng'}
      activeClassName="usa-current"
      data-cy={'nav-link-public-engagement'}>
      {intl.formatMessage(COMMON_COPY.HEADER.PUBLIC_ENG)}
    </Link>,
    <Link
      to={PAGES_ENDPOINTS.FAQS}
      key={'faqs'}
      activeClassName="usa-current"
      data-cy={'nav-link-faqs'}>
      {intl.formatMessage(COMMON_COPY.HEADER.FAQS)}
    </Link>,
  ];

  // Methodology & Data Nav component
  const MethNav = () =>
    <>
      {/* Add a className of usa-current anytime this component renders when the location of the app is on
      the Methodology page or the downloads page. This will style the nav link with a bottom border */}
      <NavDropDownButton
        className={
          lastSegmentLocation === PAGES_ENDPOINTS.METHODOLOGY.slice(1) ||
          lastSegmentLocation === PAGES_ENDPOINTS.DOWNLOADS.slice(1) ?
          'usa-current' :
          ''
        }
        key="methDropDown"
        label={intl.formatMessage(COMMON_COPY.HEADER.METHODOLOGY)}
        menuId="methMenu"
        isOpen={isOpen[0]}
        onToggle={(): void => onToggle(0)}
        data-cy={'nav-dropdown-methodology'}
      >
      </NavDropDownButton>
      <Menu
        id='methMenu'
        type='subnav'
        items={methPageSubNavLinks}
        isOpen={isOpen[0]}
      >
      </Menu>
    </>;

  // About Nav component
  const AboutNav = () =>
    <>
      {/* Add a className of usa-current anytime this component renders when the location of the app is on
      the About, FAQS or Public Eng page. This will style the nav link with a bottom border */}
      <NavDropDownButton
        className={
          lastSegmentLocation === PAGES_ENDPOINTS.ABOUT.slice(1) ||
          lastSegmentLocation === PAGES_ENDPOINTS.FAQS.slice(1) ||
          lastSegmentLocation === PAGES_ENDPOINTS.PUBLIC_ENG.slice(1) ?
          'usa-current' :
          ''
        }
        key="aboutDropDown"
        label={intl.formatMessage(COMMON_COPY.HEADER.ABOUT)}
        menuId="aboutMenu"
        isOpen={isOpen[1]}
        onToggle={(): void => onToggle(1)}
        data-cy={'nav-dropdown-about'}
      >
      </NavDropDownButton>
      <Menu
        id='aboutMenu'
        type='subnav'
        items={aboutPageSubNavLinks}
        isOpen={isOpen[1]}
      >
      </Menu>
    </>;

  // Navigation links for app
  const navLinks = [
    <Link
      to={PAGES_ENDPOINTS.EXPLORE}
      key={'explore-map'}
      activeClassName="usa-current"
      data-cy={'nav-link-explore-the-map'}>
      {intl.formatMessage(COMMON_COPY.HEADER.EXPLORE)}
    </Link>,
    <MethNav key="methDropDown"/>,
    <AboutNav key="aboutDropDown"/>,
    <Link
      to={PAGES_ENDPOINTS.CONTACT}
      key={'contact'}
      activeClassName="usa-current"
      data-cy={'nav-link-contact'}>
      {intl.formatMessage(COMMON_COPY.HEADER.CONTACT)}
    </Link>,
    // Temporarily removing language link until translation is completed
    // <div key={'language'}>
    //   <Language isDesktop={false}/>
    // </div>,
  ];

  return (
    <Header basic={true} role={'banner'}>

      {/* Banners */}
      <GovernmentBanner />
      <BetaBanner/>

      {/* Logo and Navigation */}
      <J40MainGridContainer>

        <Grid className={styles.logoNavRow} row>

          {/* Logo */}
          <Grid col={1}>
            <Link
              to={PAGES_ENDPOINTS.EXPLORE}
              key={'explore-map'}
              data-cy={'nav-link-explore-the-map'}
            >
              <img className={styles.logo} src={siteLogo} alt={`${logoLine1}`} />
            </Link>
          </Grid>

          {/* Logo Title */}
          <Grid col={6}>
            <Link
              to={PAGES_ENDPOINTS.EXPLORE}
              key={'explore-map'}
              className="remove-link-style"
              data-cy={'nav-link-explore-the-map'}
            >
              <div className={styles.logoTitle}>
                {logoLine1}
              </div>
            </Link>
          </Grid>

          {/* Nav links */}
          <Grid col={'fill'} className={styles.navLinks}>

            <NavMenuButton
              key={'mobileMenuButton'}
              onClick={toggleMobileNav}
              label="Menu">
            </NavMenuButton>
            <PrimaryNav
              items={navLinks}
              mobileExpanded={mobileNavOpen}
              onToggleMobileNav={toggleMobileNav}
            >
            </PrimaryNav>
          </Grid>

        </Grid>
      </J40MainGridContainer>

      {/* Alert */}
      {<J40MainGridContainer>
        {/* <Alert
          className={styles.alert}
          type="info"
          heading={intl.formatMessage(COMMON_COPY.ALERTS.ALERT_2_TITLE.TITLE)}
          headingLevel={'h1'}>
          {COMMON_COPY.ALERTS.ALERT_2_DESCRIPTION}
        </Alert> */}

        <Alert
          className={styles.alert}
          type="info"
          heading={intl.formatMessage(COMMON_COPY.ALERTS.ALERT_1_TITLE.TITLE)}
          headingLevel={'h1'}>
          {COMMON_COPY.ALERTS.ALERT_1_DESCRIPTION}
        </Alert>
      </J40MainGridContainer>
      }
    </Header>
  );
};

export default J40Header;
