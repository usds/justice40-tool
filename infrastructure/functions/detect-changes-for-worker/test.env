REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1