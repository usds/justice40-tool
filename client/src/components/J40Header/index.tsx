import J40Header from './J40Header';

export default J40Header;
