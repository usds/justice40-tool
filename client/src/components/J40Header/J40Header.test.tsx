import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import J40Header from './J40Header';

describe('rendering of the J40Header', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <J40Header location={location}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
