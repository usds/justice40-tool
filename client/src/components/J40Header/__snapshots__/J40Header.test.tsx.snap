// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the J40Header checks if component renders 1`] = `
<DocumentFragment>
  <header
    class="usa-header usa-header--basic"
    data-testid="header"
    role="banner"
  >
    <div>
      <div>
        <section
          class="usa-banner"
          data-testid="govBanner"
        >
          <div
            class="usa-accordion"
          >
            <header
              class="usa-banner__header"
            >
              <div
                class="usa-banner__inner"
              >
                <div
                  class="grid-col-auto"
                >
                  <img
                    alt="U.S. flag"
                    class="usa-banner__header-flag"
                    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAALCAMAAABBPP0LAAAAG1BMVEUdM7EeNLIeM7HgQCDaPh/bPh/bPx/////bPyBEby41AAAAUElEQVQI123MNw4CABDEwD3jC/9/MQ1BQrgeOSkIqYe2o2FZtthXgQLgbHVMZdlsfUQFQnHtjP1+8BUhBDKOqtmfot6ojqPzR7TjdU+f6vkED+IDPhTBcMAAAAAASUVORK5CYII="
                  />
                </div>
                <div
                  class="grid-col-fill tablet:grid-col-auto"
                >
                  <p
                    class="usa-banner__header-text"
                  >
                    An official website of the United States government
                  </p>
                  <p
                    aria-hidden="true"
                    class="usa-banner__header-action"
                  >
                    Here’s how you know
                  </p>
                </div>
                <button
                  aria-controls="gov-banner"
                  aria-expanded="false"
                  class="usa-accordion__button usa-banner__button"
                  type="button"
                >
                  <span
                    class="usa-banner__button-text"
                  >
                    Here’s how you know
                  </span>
                </button>
              </div>
            </header>
            <div
              class="usa-banner__content usa-accordion__content"
              hidden=""
              id="gov-banner"
            >
              <div
                class="grid-row grid-gap-lg"
              >
                <div
                  class="usa-banner__guidance tablet:grid-col-6"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="usa-banner__icon usa-media-block__img"
                    role="img"
                    src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0IiB2aWV3Qm94PSIwIDAgNjQgNjQiPjx0aXRsZT5pY29uLWRvdC1nb3Y8L3RpdGxlPjxwYXRoIGZpbGw9IiMyMzc4QzMiIGZpbGwtcnVsZT0iZXZlbm9kZCIgZD0iTTMyIDBjMTcuNjczIDAgMzIgMTQuMzI3IDMyIDMyIDAgMTcuNjczLTE0LjMyNyAzMi0zMiAzMkMxNC4zMjcgNjQgMCA0OS42NzMgMCAzMiAwIDE0LjMyNyAxNC4zMjcgMCAzMiAwem0wIDEuMjA4QzE0Ljk5NCAxLjIwOCAxLjIwOCAxNC45OTQgMS4yMDggMzJTMTQuOTk0IDYyLjc5MiAzMiA2Mi43OTIgNjIuNzkyIDQ5LjAwNiA2Mi43OTIgMzIgNDkuMDA2IDEuMjA4IDMyIDEuMjA4em0xMC41OSAzOC44NThhLjg1Ny44NTcgMCAwIDEgLjg4Mi44MjJ2MS42NDJIMTguODg2di0xLjY0MmEuODU3Ljg1NyAwIDAgMSAuODgyLS44MjJINDIuNTl6TTI1LjQ0MyAyNy43NzR2OS44MjloMS42NDJ2LTkuODNoMy4yNzN2OS44M0gzMnYtOS44M2gzLjI3MnY5LjgzaDEuNjQzdi05LjgzaDMuMjcydjkuODNoLjc2YS44NTcuODU3IDAgMCAxIC44ODIuODIxdi44MjFoLTIxLjN2LS44MDlhLjg1Ny44NTcgMCAwIDEgLjg4LS44MmguNzYydi05Ljg0MmgzLjI3MnptNS43MzYtOC4xODhsMTIuMjkzIDQuOTE1djEuNjQyaC0xLjYzYS44NTcuODU3IDAgMCAxLS44ODIuODIySDIxLjQxYS44NTcuODU3IDAgMCAxLS44ODItLjgyMmgtMS42NDJ2LTEuNjQybDEyLjI5My00LjkxNXoiLz48L3N2Zz4="
                  />
                  <div
                    class="usa-media-block__body"
                  >
                    <p>
                      <strong>
                        Official websites use .gov
                      </strong>
                      <br />
                      A 
                      <strong>
                        .gov
                      </strong>
                       website belongs to an official government organization in the United States.
                    </p>
                  </div>
                </div>
                <div
                  class="usa-banner__guidance tablet:grid-col-6"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="usa-banner__icon usa-media-block__img"
                    role="img"
                    src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0IiB2aWV3Qm94PSIwIDAgNjQgNjQiPjx0aXRsZT5pY29uLWh0dHBzPC90aXRsZT48cGF0aCBmaWxsPSIjNzE5RjJBIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik0zMiAwYzE3LjY3MyAwIDMyIDE0LjMyNyAzMiAzMiAwIDE3LjY3My0xNC4zMjcgMzItMzIgMzJDMTQuMzI3IDY0IDAgNDkuNjczIDAgMzIgMCAxNC4zMjcgMTQuMzI3IDAgMzIgMHptMCAxLjIwOEMxNC45OTQgMS4yMDggMS4yMDggMTQuOTk0IDEuMjA4IDMyUzE0Ljk5NCA2Mi43OTIgMzIgNjIuNzkyIDYyLjc5MiA0OS4wMDYgNjIuNzkyIDMyIDQ5LjAwNiAxLjIwOCAzMiAxLjIwOHptMCAxOC44ODZhNy4yNDUgNy4yNDUgMCAwIDEgNy4yNDUgNy4yNDV2My4xMDNoLjUyYy44NiAwIDEuNTU3LjY5OCAxLjU1NyAxLjU1OHY5LjMyMmMwIC44Ni0uNjk3IDEuNTU4LTEuNTU3IDEuNTU4aC0xNS41M2MtLjg2IDAtMS41NTctLjY5Ny0xLjU1Ny0xLjU1OFYzMmMwLS44Ni42OTctMS41NTggMS41NTctMS41NThoLjUyVjI3LjM0QTcuMjQ1IDcuMjQ1IDAgMCAxIDMyIDIwLjA5NHptMCAzLjEwM2E0LjE0MiA0LjE0MiAwIDAgMC00LjE0MiA0LjE0MnYzLjEwM2g4LjI4NFYyNy4zNEE0LjE0MiA0LjE0MiAwIDAgMCAzMiAyMy4xOTd6Ii8+PC9zdmc+"
                  />
                  <div
                    class="usa-media-block__body"
                  >
                    <p>
                      <strong>
                        Secure .gov websites use HTTPS
                      </strong>
                      <br />
                      A 
                      <strong>
                        lock (
                        <span
                          class="icon-lock"
                        >
                          <img
                            alt="lock"
                            class="usa-banner__lock-image"
                            role="img"
                            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjUyIiBoZWlnaHQ9IjY0IiB2aWV3Qm94PSIwIDAgNTIgNjQiPjx0aXRsZT5sb2NrPC90aXRsZT48cGF0aCBmaWxsPSIjMUIxQjFCIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik0yNiAwYzEwLjQ5MyAwIDE5IDguNTA3IDE5IDE5djloM2E0IDQgMCAwIDEgNCA0djI4YTQgNCAwIDAgMS00IDRINGE0IDQgMCAwIDEtNC00VjMyYTQgNCAwIDAgMSA0LTRoM3YtOUM3IDguNTA3IDE1LjUwNyAwIDI2IDB6bTAgOGMtNS45NzkgMC0xMC44NDMgNC43Ny0xMC45OTYgMTAuNzEyTDE1IDE5djloMjJ2LTljMC02LjA3NS00LjkyNS0xMS0xMS0xMXoiLz48L3N2Zz4="
                            title="Lock"
                          />
                        </span>
                        )
                      </strong>
                       or 
                      <strong>
                        https://
                      </strong>
                       means you’ve safely connected to the .gov website. Share sensitive information only on official, secure websites.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div>
      <div>
        <strong>
          This tool has been updated.
        </strong>
         The 1.0 version of the tool was released on Nov 22, 2022.
      </div>
    </div>
    <div
      class="grid-container-desktop-lg"
      data-testid="gridContainer"
    >
      <div
        class="grid-row"
        data-testid="grid"
      >
        <div
          class="grid-col-1"
          data-testid="grid"
        >
          <a
            data-cy="nav-link-explore-the-map"
            href="/en/"
          >
            <img
              alt="Climate and Economic Justice Screening Tool"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="grid-col-6"
          data-testid="grid"
        >
          <a
            class="remove-link-style"
            data-cy="nav-link-explore-the-map"
            href="/en/"
          >
            <div>
              Climate and Economic Justice Screening Tool
            </div>
          </a>
        </div>
        <div
          class="grid-col-fill"
          data-testid="grid"
        >
          <button
            class="usa-menu-btn"
            data-testid="navMenuButton"
            type="button"
          >
            Menu
          </button>
          <nav
            class="usa-nav"
          >
            <button
              class="usa-nav__close"
              data-testid="navCloseButton"
              type="button"
            >
              <img
                alt="close"
                src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0IiB2aWV3Qm94PSIwIDAgNjQgNjQiPjx0aXRsZT5jbG9zZTwvdGl0bGU+PHBhdGggZmlsbD0iIzU2NUM2NSIgZmlsbC1ydWxlPSJldmVub2RkIiBkPSJNNTcuMDQyIDEuMTVsNS44MDkgNS44MDhhNCA0IDAgMCAxIDAgNS42NTdMNDMuNDY1IDMybDE5LjM4NiAxOS4zODVhNCA0IDAgMCAxIDAgNS42NTdsLTUuODA5IDUuODA5YTQgNCAwIDAgMS01LjY1NyAwTDMyIDQzLjQ2NSAxMi42MTUgNjIuODUxYTQgNCAwIDAgMS01LjY1NyAwbC01LjgwOS01LjgwOWE0IDQgMCAwIDEgMC01LjY1N0wyMC41MzUgMzIgMS4xNDkgMTIuNjE1YTQgNCAwIDAgMSAwLTUuNjU3bDUuODA5LTUuODA5YTQgNCAwIDAgMSA1LjY1NyAwTDMyIDIwLjUzNSA1MS4zODUgMS4xNDlhNCA0IDAgMCAxIDUuNjU3IDB6Ii8+PC9zdmc+"
              />
            </button>
            <ul
              class="usa-nav__primary usa-accordion"
            >
              <li
                class="usa-nav__primary-item"
              >
                <a
                  data-cy="nav-link-explore-the-map"
                  href="/en/"
                >
                  Explore the map
                </a>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  aria-controls="methMenu"
                  aria-expanded="false"
                  class="usa-accordion__button usa-nav__link"
                  data-cy="nav-dropdown-methodology"
                  data-testid="navDropDownButton"
                  type="button"
                >
                  <span>
                    Methodology & data
                  </span>
                </button>
                <ul
                  class="usa-nav__submenu"
                  hidden=""
                  id="methMenu"
                >
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-methodology"
                      href="/en/methodology"
                    >
                      Methodology & data
                    </a>
                  </li>
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-downloads"
                      href="/en/downloads"
                    >
                      Downloads
                    </a>
                  </li>
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-previous-versions"
                      href="/en/previous-versions"
                    >
                      Previous versions
                    </a>
                  </li>
                </ul>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  aria-controls="aboutMenu"
                  aria-expanded="false"
                  class="usa-accordion__button usa-nav__link"
                  data-cy="nav-dropdown-about"
                  data-testid="navDropDownButton"
                  type="button"
                >
                  <span>
                    About
                  </span>
                </button>
                <ul
                  class="usa-nav__submenu"
                  hidden=""
                  id="aboutMenu"
                >
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-about"
                      href="/en/about"
                    >
                      About
                    </a>
                  </li>
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-public-engagement"
                      href="/en/public-engagement"
                    >
                      Engagement calendar
                    </a>
                  </li>
                  <li
                    class="usa-nav__submenu-item"
                  >
                    <a
                      data-cy="nav-link-faqs"
                      href="/en/frequently-asked-questions"
                    >
                      Frequently asked questions
                    </a>
                  </li>
                </ul>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <a
                  data-cy="nav-link-contact"
                  href="/en/contact"
                >
                  Contact
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <div
      class="grid-container-desktop-lg"
      data-testid="gridContainer"
    >
      <div
        class="usa-alert usa-alert--info"
        data-testid="alert"
      >
        <div
          class="usa-alert__body"
        >
          <h1
            class="usa-alert__heading"
          >
            Version 1.0 of the tool is now available
          </h1>
          <p
            class="usa-alert__text"
          >
            The Council on Environmental Quality (CEQ) made the version 1.0 of the tool available on 11/22/2022. For more information about the improvements to the tool, read CEQ's 
            <a
              class="usa-link usa-link--external"
              data-cy=""
              href="https://www.whitehouse.gov/ceq/news-updates/2022/11/22/biden-harris-administration-launches-version-1-0-of-climate-and-economic-justice-screening-tool-key-step-in-implementing-president-bidens-justice40-initiative/"
              rel="noreferrer"
              target="_blank"
            >
              press release
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
